### PR TITLE
drivers: flash: stm32 qspi driver reads or writes non-null size

### LINKED ISSUE #42417: tests drivers flash on stm32 qspi controller
**Describe the bug**
when testing the stm32 flash driver on the qspi nor-flash controller,  typically disco_l475_iot1, the test does not exit from flash_stm32_qspi_read. 

**To Reproduce**
1. delete the target overlay to enable the qspi node: tests/drivers/flash/boards/disco_l475_iot1.overlay
2. west build -p auto -b disco_l475_iot1 tests/drivers/flash
3. west flash

**Impact**
test fails on this target board

**Logs and console output**
```
I: MX25R6435F: SFDP v 1.6 AP ff with 3 PH                                       
I: PH0: ff00 rev 1.6: 16 DW @ 30                                                
I: MX25R6435F: 8 MiBy flash                                                     
I: PH1: ffc2 rev 1.0: 4 DW @ 110                                                
I: Device MX25R6435F initialized                                                
*** Booting Zephyr OS build v3.0.0-rc1-93-g0343b67ba4ed  ***                    
Running test suite flash_driver_test                                            
===================================================================             
START - test_setup                                                              
 PASS - test_setup in 0.75 seconds                                              
===================================================================             
START - test_read_unaligned_address
```

**Environment (please complete the following information):**
zephyr OS 833d6c2315b4b3c2bf6a4ada8b3e28fb4292ef00

### PR DESCRIPTION
**Pull request by FRASTM**
In case the length of data to read/write/erase is null, the qspi drivers immediately returns without sending any command to the QSPI controller.
This is the same approach as for the stm32 flash driver flash_stm32.c.

The tests/drivers/flash can successfully run on the stm32l4r5 disco kit.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/42417

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
$ west build -p auto -b disco_l475_iot1 tests/drivers/flash
```
I: MX25R6435F: SFDP v 1.6 AP ff with 3 PH
I: PH0: ff00 rev 1.6: 16 DW @ 30
I: MX25R6435F: 8 MiBy flash
I: PH1: ffc2 rev 1.0: 4 DW @ 110
I: Device MX25R6435F initialized
*** Booting Zephyr OS build v3.0.0-rc1-93-g5dc63b016c08  ***
Running test suite flash_driver_test
===================================================================
START - test_setup
 PASS - test_setup in 0.75 seconds
===================================================================
START - test_read_unaligned_address
 PASS - test_read_unaligned_address in 0.33 seconds
===================================================================
Test suite flash_driver_test succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

**Review by erwango — CHANGES_REQUESTED** [review]
Please also delete `tests/drivers/flash/boards/stm32f746g_disco.overlay`

**Comment by FRASTM** [comment]
idem for stm32f746g_disco : run the tests/drivers/flash/ on the QSPI nor-flash

**Review by erwango — APPROVED** [review]

File: drivers/flash/flash_stm32_qspi.c [hunk 1]
```diff
@@ -311,6 +311,11 @@ static int flash_stm32_qspi_read(const struct device *dev, off_t addr,
 		return -EINVAL;
 	}
 
+	/* read non-zero size */
+	if (!size) {
```
Comment (gmarull) [inline drivers/flash/flash_stm32_qspi.c:315]: nit: better use essentially boolean expressions:

```suggestion
	if (size == 0U) {
```

File: drivers/flash/flash_stm32_qspi.c [hunk 2]
```diff
@@ -359,6 +364,11 @@ static int flash_stm32_qspi_write(const struct device *dev, off_t addr,
 		return -EINVAL;
 	}
 
+	/* write non-zero size */
+	if (!size) {
```
Comment (gmarull) [inline drivers/flash/flash_stm32_qspi.c:368]:
```suggestion
	if (size == 0U) {
```

File: drivers/flash/flash_stm32_qspi.c [hunk 3]
```diff
@@ -429,6 +439,11 @@ static int flash_stm32_qspi_erase(const struct device *dev, off_t addr,
 		return -EINVAL;
 	}
 
+	/* erase non-zero size */
+	if (!size) {
```
Comment (gmarull) [inline drivers/flash/flash_stm32_qspi.c:443]:
```suggestion
	if (size == 0U) {
```

Comment (FRASTM) [reply drivers/flash/flash_stm32_qspi.c]: changed

**Review by gmarull — APPROVED** [review]

**Comment by FRASTM** [comment]
changed to compare size with 0 explicitely.
Note that flash_stm32.c has the `if (!len) {`

**Pull request merged**